### PR TITLE
Renovation: implement button's iconPosition

### PR DESCRIPTION
**Pull request by LazyLahtak**

### CODE REVIEW

File: js/renovation/button.tsx [hunk 1]
```diff
@@ -5,12 +5,14 @@ import Widget from './widget';
 
 const WidgetJSX = JSXConstructor<Widget>(Widget);
 
-const getImageContainerJSX = (source: string) => {
+const getImageContainerJSX = (source: string, position: string = '') => {
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: You have already defined the `position` property in the definition.

BTW, we should avoid defining defaults to function arguments.

File: js/renovation/button.tsx [hunk 2]
```diff
@@ -5,12 +5,14 @@ import Widget from './widget';
...
+    const ICON_RIGHT_CLASS = position !== 'left' ? 'dx-icon-right' : '';
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: It's not a constant. Why are we using capital letters?

File: js/renovation/button.tsx [hunk 3]
```diff
@@ -29,8 +31,12 @@ const getCssClasses = (model: any) => {
     text && classNames.push('dx-button-has-text');
     icon && classNames.push('dx-button-has-icon');
 
+    if (iconPosition !== 'left') {
+        classNames.push('dx-button-icon-right');
+    }
+
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]:
```js
iconPosition !== 'left' && classNames.push('dx-button-icon-right');
```

File: js/renovation/button.tsx [hunk 4]
```diff
@@ -82,8 +88,9 @@ export const viewFunction = (viewModel: Button) => (
             </div>
         ) || (
             <div className="dx-button-content">
-                {viewModel.icon}
+                {viewModel.iconPosition === 'left' && viewModel.icon}
                 {viewModel.text && <span className="dx-button-text">{viewModel.text}</span>}
+                {viewModel.iconPosition !== 'left' && viewModel.icon}
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: We can extract the next condition into a variable `viewModel.iconPosition === 'left'`

File: testing/jest/button.tests.tsx [hunk 5]
```diff
@@ -79,6 +79,39 @@ describe('Button', () => {
...
+                    iconPosition: 'right',
+                });
+
+                const content = button.find('.dx-button-content');
+
+                expect(button.hasClass('dx-button-icon-right'))
+                    .toBeTruthy();
+                expect(content.children().at(0).is('.dx-button-text'))
+                    .toBeTruthy();
+                expect(content.children().at(1).is('.dx-icon.dx-icon-test.dx-icon-right'))
```
Comment (MaximKudriavtsev) [inline testing/jest/button.tests.tsx]: We can extract `content.children()` and the same above.

File: js/renovation/button.tsx [hunk 6]
```diff
@@ -99,9 +106,10 @@ export const viewFunction = (viewModel: Button) => (
 export default class Button extends Widget {
     @Prop() classNames?: string[];
     @Prop() contentRender?: any;
-    @Prop() icon?: string;
+    @Prop() icon?: string = '';
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: Why we defined it?

BTW, if we have a default option here, we should have the same default prop in the `**.j.jsx` file.

File: testing/jest/button.tests.tsx [hunk 7]
```diff
@@ -79,6 +79,39 @@ describe('Button', () => {
                 expect(buttonContentChildren.render().text()).toBe('My button123');
             });
         });
+
+        describe('iconPosition', () => {
+            it('should render icon before text if iconPosition is left', () => {
```
Comment (MaximKudriavtsev) [inline testing/jest/button.tests.tsx]: Also, the default behavior should be tested. Or what right default properties are defined.

**Review by MaximKudriavtsev — DISMISSED** [review]

**Review by MaximKudriavtsev — DISMISSED** [review]

**Review by MaximKudriavtsev — DISMISSED** [review]

**Comment by LazyLahtak** [comment]
Ready to review

**Review by churkin — APPROVED** [review]

**Pull request merged**